x/evm/keeper/msg_server.go

```diff
@@ -0,0 +1,187 @@
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/sei-protocol/sei-chain/x/evm/state"
+	"github.com/sei-protocol/sei-chain/x/evm/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl returns an implementation of the MsgServer interface
+// for the provided Keeper.
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+var _ types.MsgServer = msgServer{}
+
+func (server msgServer) EVMTransaction(goCtx context.Context, msg *types.MsgEVMTransaction) (*types.MsgEVMTransactionResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	stateDB := state.NewStateDBImpl(ctx, &server)
+	tx, _ := msg.AsTransaction()
+	ctx, gp := server.getGasPool(ctx)
+	emsg, err := server.getEVMMessage(ctx, tx)
+	if err != nil {
+		return &types.MsgEVMTransactionResponse{}, err
+	}
+
+	res, err := server.applyEVMMessage(ctx, emsg, stateDB, gp)
+	if err != nil {
+		return &types.MsgEVMTransactionResponse{}, err
+	}
+
+	if err := server.writeReceipt(ctx, tx, emsg, res); err != nil {
+		return &types.MsgEVMTransactionResponse{}, err
```

Comment:
what does a writeReceipt do? Coming from very ltitle evm knowlege, some more docstrings in general (at least for each of the `new` classes, may help

Reply:
yeah, added in the keeper files